### PR TITLE
p2p: add test for pqueue dequeue full error

### PR DESCRIPTION
**Pull request by williambanfield**
This adds a test for closing the `pqueue` while the `pqueue` contains data that has not yet been dequeued.

This issue was found while debugging #6705 

This test will fail until @cmwaters fix for this condition is merged.

### CODE REVIEW

File: internal/p2p/pqueue_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+func TestCloseWhileDequeueFull(t *testing.T) {
+	enqueueLength := 5
+	chDescs := []ChannelDescriptor{
+		{ID: 0x01, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x02, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x03, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x04, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x05, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x06, Priority: 1, MaxSendBytes: 4},
```
Comment (cmwaters) [inline internal/p2p/pqueue_test.go:18]:
```suggestion
		{ID: 0x01, Priority: 1, MaxSendBytes: 4},
```
nit: you could just have a single channel

**Review by cmwaters — APPROVED** [review]
LGTM

File: internal/p2p/pqueue_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		pqueue.close()
+		close(doneCh)
+	}()
+
+	select {
+	case <-doneCh:
+	case <-time.After(2 * time.Second):
+		t.Fatal("pqueue failed to close")
+	}
+
```
Comment (alexanderbez) [inline internal/p2p/pqueue_test.go:44]:
```suggestion
```

File: internal/p2p/pqueue_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		pqueue.enqueue() <- Envelope{
+			channelID: 0x01,
+			Message:   &testMessage{Value: "foo"}, // 5 bytes
+		}
+	}
+
+	go pqueue.process()
+
+	// sleep to allow context switch for process() to run
+	time.Sleep(10 * time.Millisecond)
```
Comment (alexanderbez) [inline internal/p2p/pqueue_test.go:32]: Wish there was a cleaner way to know process() is executing. Perhaps we could add a status field or something to the queue. Not really a blocker, just wanting to point it out.

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**